src/Modules/Visualization/ShowField.cc

```diff
@@ -240,81 +241,45 @@ GeometryHandle ShowFieldModule::buildGeometryObject(
   const int dim = field->vmesh()->dimensionality();
   if (showEdges && dim < 1) { showEdges = false; }
   if (showFaces && dim < 2) { showFaces = false; }
+  
+  //adjust color map to scalar data if there is one.
+  ColorMap *cmap = colorMap.get().get();
+  ColorMap adjustedCMap(cmap->getColorMapName(),cmap->getColorMapResolution(),
+                        cmap->getColorMapShift(), cmap->getColorMapInvert(),
+                        cmap->getColorMapRescaleScale(), cmap->getColorMapRescaleShift());
+  if (colorMap) {
```

Comment:
On line 246 you've already dereferenced the colorMap pointer, so this check should happen earlier.